### DISCUSSION TITLE
Not working after install

### DISCUSSION THREAD

**Discussion by SanChriTa** [teslamate-org/teslamate#2897] [Q&A] [answered]
Hi,
I'm not familiar with docker and I'm having a lot of trouble setting it up.
The installation is going according to: https://www.tslaspot.com/teslamate-setup/#install except that I have my own server (ubuntu 22.04 LTS) on the local network. I don't book anything externally (including domain, dns, etc). 
In the yaml file in the FQDN where there is a domain like grafana.example.com, should I enter my internal IP of this server without port and for both teslamate and grafana the same?
I would like to open teslamate and grafane via the internal ip by adding the port number at the end setting it beforehand (preferably two digits). Going to the ip without port there is a 404 error, and typing the ports I see in cmd I get nothing. 
What needs to be done here? Is it a matter of setting something with the network in docker?

![docker-tm](https://user-images.githubusercontent.com/113474559/190004215-78689465-b004-46d9-a65c-60a60dfc27a3.jpg)

**Comment by cwanja** [comment, marked as answer]
You are following a guide that looks to walk you through exposing it to the internet. If, as you said, you do not need anything externally, I would follow this guide here:
https://docs.teslamate.org/docs/installation/docker

That will install it on Docker and only accessible inside of your network.

**Reply by SanChriTa** [reply]
Installation without errors, but entering the ip of the machine and adding port both 3000 and 4000 - connection timeout exceeded
![docker-tm2](https://user-images.githubusercontent.com/113474559/190008170-efb3a676-bb10-4fa6-929a-22cf960b5d49.jpg)

**Reply by cwanja** [reply]
What is in the logs? To retrieve the TeslaMate Docker container logs, run `docker-compose logs teslamate`

**Reply by SanChriTa** [reply]
![docker-tm3](https://user-images.githubusercontent.com/113474559/190008816-679e329d-070c-4e90-8e89-6d8f0a8b86b3.jpg)

**Reply by cwanja** [reply]
What is in the database logs?

**Reply by SanChriTa** [reply]
I removed the previous images via docker system prune. Maybe something didn't delete correctly? Indeed, in the first screenshot before changing the yml version next to postgres there was a status of active container, and here not.

![docker-tm4](https://user-images.githubusercontent.com/113474559/190010323-d3fa356a-30d9-4f83-b2d6-3f9d1d1324a8.jpg)

**Reply by cwanja** [reply]
Definitely looks like you have a conflicting database per the errors. I would remove any lingering databases that were created. Here is one option from a quick Google search.
https://stackoverflow.com/questions/58066879/how-to-delete-postgres-database-within-docker-container

**Reply by SanChriTa** [reply]
thanks,
I don't know what caused these errors, but after a second reinstallation of linux it finally worked.
Only it doesn't ask for the password even though the .htpasswd file is created. This must have been in the yml before? Can it still be added?

**Reply by cwanja** [reply]
It is because you had conflicting database versions. First installation was v12. Second was v14 and the database was lingering around. Not sure if you can add a .htpassword, but you can mess with it by adding a proxy service and securing the 4000 port. 

Any reason you want to secure the page since it is not exposed to the internet?

**Reply by SanChriTa** [reply]
I noticed, but somehow couldn't remove the previous ones. Maybe I even marked when installing Linux to have postresql already some I forgot about.

The reason is that other users still have access to this subnet and it is imperative that I secure it. Also, if I lend my car to someone, by going into the favorites tab someone has full access to my driving data. The car will probably be equipped with a router that will be permanently connected via an OpenVPN client to the local network.
I would absolutely like to secure this somehow. Could you suggest something?

Your advice has helped. Thank you.

EDIT: https://www.tslaspot.com/teslamate-setup/
Here it is a password so maybe just add something to the yaml file to include this .htpasswd file in the method you provided earlier?
